platform-tests/src/test/java/org/junit/platform/engine/support/hierarchical/ForkJoinPoolHierarchicalTestExecutorServiceTests.java

```diff
@@ -67,44 +93,128 @@ void defersTasksWithIncompatibleLocks() throws Exception {
 			deferred.countDown();
 		};
 
-		var isolatedTask = new DummyTestTask("isolatedTask", globalReadWriteLock,
-			t -> threadNamesByTaskIdentifier.put(t.identifier(), Thread.currentThread().getName()));
+		var incompatibleTask = new DummyTestTask("incompatibleTask", incompatibleLock);
+
+		var tasks = runWithAttemptedWorkStealing(taskEventListener, incompatibleTask, initialLock, () -> {
+			try {
+				deferred.await();
+			}
+			catch (InterruptedException e) {
+				System.out.println("Interrupted while waiting for task to be deferred");
+			}
+		});
+
+		assertEquals(incompatibleTask, deferredTask.get());
+		assertEquals(tasks.get("nestedTask").threadName, tasks.get("leafTask2").threadName);
+		assertNotEquals(tasks.get("leafTask1").threadName, tasks.get("leafTask2").threadName);
+	}
+
+	static List<Arguments> compatibleLockCombinations() {
+		return List.of(//
+			arguments(//
+				Set.of(GLOBAL_READ, new ExclusiveResource("a", LockMode.READ)), //
+				Set.of(GLOBAL_READ, new ExclusiveResource("b", LockMode.READ)) //
+			), //
+			arguments(//
+				Set.of(GLOBAL_READ, new ExclusiveResource("a", LockMode.READ_WRITE)), //
+				Set.of(GLOBAL_READ, new ExclusiveResource("b", LockMode.READ_WRITE)) //
+			), //
+			arguments(//
+				Set.of(GLOBAL_READ, new ExclusiveResource("a", LockMode.READ_WRITE)), //
+				Set.of(GLOBAL_READ, new ExclusiveResource("a", LockMode.READ)) //
+			), //
+			arguments(//
+				Set.of(GLOBAL_READ_WRITE), //
+				Set.of(GLOBAL_READ) //
+			)//
```

Comment:
:x: these are wrong and break the guarantees we want to ensure with our `ExclusiveResource` in the first place.
You could argue that `a`,`b` RW is fine, but I would say we should not mix RW locks.

Reply:
Which guarantees do these break?

Reply:
As discussed elsewhere, as soon as one side is holding a RW lock, we can't allow any additional locks to avoid breaking the exclusiveness guarantees.